### PR TITLE
[OSCI][FIX] Discover: Fix the Subfields Display Issue

### PR DESCRIPTION
**Pull request by MadaniKK**
### Description

Fix the subfields display issue in Discover left nav bar.

### Issues Resolved
Issue #5400

## Screenshot
Before:
image

After:
image

### Check List

- [☑️ ] All tests pass
  - [ ☑️ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [☑️  ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ananzh** [comment]
This is great! Just need a changelog update.

**Comment by MadaniKK** [comment]
> This is great! I like all the discussions in the original issue and glad that you find such a simple yet smart solution. Just need a changelog update.

Thank you for supporting me thru all the digging!

**Comment by Willie-The-Lord** [comment]
Hi @MadaniKK, it seems like your code editor auto format `CHANGELOG.md` when you are saving the file.

**Comment by MadaniKK** [comment]
> Hi @MadaniKK, it seems like your code editor auto format `CHANGELOG.md` when you are saving the file.

Ah good catch, maybe I should redo the add changelog?

**Comment by Willie-The-Lord** [comment]
Yeah, me personally will use `git revert` [the commit](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5429/commits/8d64fad4d60a6e71ffa7d421a44195414077c77f) and push back to the remote. Hope it helps!

**Comment by Willie-The-Lord** [comment]
And if you are using vscode, you can find `Save without Formatting` in the Command Palette.
Screenshot 2023-11-03 at 9 14 23 PM

**Comment by MadaniKK** [comment]
> And if you are using vscode, you can find `Save without Formatting` in the Command Palette. Screenshot 2023-11-03 at 9 14 23 PM



> And if you are using vscode, you can find `Save without Formatting` in the Command Palette. Screenshot 2023-11-03 at 9 14 23 PM

I just tried this and It's still causing the same amount of change to the changelog.md ` 1 file changed, 9 insertions(+), 8 deletions(-)`

**Comment by Willie-The-Lord** [comment]
Did you close the `COMMIT_EDIT_MSG` file?  the revert process is completed when the file is closed.

**Comment by AMoo-Miki** [comment]
The previous commits were signed-off by Qiwen Li; I will manually pass DCO.

**Comment by AMoo-Miki** [comment]
Considering the legacy behavior, I wouldn't call this a bug fix but a very important enhancement. Thanks.

**Review by ananzh — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Add support for read-only mode through tenants ([#4498](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4498))
 - [Workspace] Add core workspace service module to enable the implementation of workspace features within OSD plugins ([#5092](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5092))
 - [Workspace] Setup workspace skeleton and implement basic CRUD API ([#5075](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5075))
+- [Discover] Display inner properties in the left navigation bar [#5429](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5429)
+
```
Comment (ananzh) [inline CHANGELOG.md:50]: nit: this line is not necessary. need to remove.

**Review by CMDWillYang — APPROVED** [review]

**Comment by ananzh** [comment]
@MadaniKK Does this also resolve https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5208 right?

I checked the issue and it seems be already fixed by this PR. Because right now, the side nav will show all the nested field, therefore user can not choose the original object. 

Screenshot 2023-11-12 at 12 57 45

**Review by ananzh — APPROVED** [review]

**Review by joshuarrrr — DISMISSED** [review]

**Review by ananzh — APPROVED** [review]

**Review by manasvinibs — APPROVED** [review]

**Pull request merged**